railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -59,18 +59,18 @@ def self.plugin_rails_options(opts, options)
 
     options[:color] = true
     options[:output_inline] = true
-    options[:patterns] = opts.order! unless run_via[:rake]
+    options[:patterns] = opts.order! unless run_via.rake?
   end
 
   def self.rake_run(patterns) # :nodoc:
-    run_via[:rake] = true
+    self.run_via = :rake if self.run_via.nil?
```

Comment:
This is because `rake_run` is called multiple times when running multiple tests in one `rake` command(e.g. `bin/rake test:models test:controllers`).